Leetcode/EigenDemo/eigen_practice.cpp

```diff
@@ -0,0 +1,128 @@
+//
+//  eigenPractice.cpp
+//  EigenDemo
+//
+//  Created by dsh on 2021/7/18.
+//
+
+#include <iostream>
+
+#include "Eigen/Eigen"
+
+#include "eigen_practice.hpp"
+
+void EigenPractice::EigenBasic() {
+    Eigen::MatrixXd m(2, 2);
+    m(0, 0) = 3;
+    m(1, 0) = 2.5;
+    m(0, 1) = -1;
+    m(1, 1) = m(1, 0) + m(0, 1);
+    std::cout << m << std::endl;
+
+    Eigen::Matrix<float, 2, 3> matrix_23;
+    Eigen::Vector3d v_3d;
+    Eigen::Matrix<float, 3, 1> vd_3d;
+    Eigen::Matrix3d matrix_33 = Eigen::Matrix3d::Zero();
+    Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> matrix_dynamic;
+    Eigen::MatrixXd matrix_x;
+
+    matrix_23 << 1, 2, 3, 4, 5, 6;
+    std::cout << "matrix 2x3 :" << std::endl;
+    std::cout << matrix_23 << std::endl;
+    std::cout << "print matrix 2x3: " << std::endl;
+    for (int i = 0; i < 2; i++) {
+        for (int j = 0; j < 3; j++)
+            std::cout << matrix_23(i, j) << "\t";
+        std::cout << std::endl;
+    }
+
+    v_3d << 3, 2, 1;
+    vd_3d << 4, 5, 6;
+
+    Eigen::Matrix<double, 2, 1> result = matrix_23.cast<double>() * v_3d;
+    std::cout << "matrix_23 * v_3d = " << result.transpose() << std::endl;
+
+    matrix_33 << 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9;
+    std::cout << "random matrix:" << std::endl;
+    std::cout << matrix_33 << std::endl;
+    std::cout << "transpose:" << std::endl;
+    std::cout << matrix_33.transpose() << std::endl;
+    std::cout << "sum: " << matrix_33.sum() << std::endl;
+    std::cout << "trace: " << matrix_33.trace() << std::endl;
+    std::cout << "times 10:" << std::endl;
+    std::cout << 10 * matrix_33 << std::endl;
+    std::cout << "inverse:" << std::endl;
+    std::cout << matrix_33.inverse() << std::endl;
+    std::cout << "det: " << matrix_33.determinant() << std::endl;
+
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix3d>
+    eigen_solver(matrix_33.transpose() * matrix_33);
+    std::cout << "Eigen values =" << std::endl;
+    std::cout << eigen_solver.eigenvalues() << std::endl;
+    std::cout << "Eigen vectors =" << std::endl;
+    std::cout << eigen_solver.eigenvectors() <<std::endl;
+
+    //matrix_NN * x = v_Nd
+    Eigen::Matrix<double, kMatrixSize, kMatrixSize> matrix_NN
+    = Eigen::MatrixXd::Random(kMatrixSize, kMatrixSize);
+    matrix_NN = matrix_NN * matrix_NN.transpose();
+    Eigen::Matrix<double, kMatrixSize, 1> v_Nd =
+    Eigen::MatrixXd::Random(kMatrixSize, 1);
+
+    clock_t time_stt = clock();
+    Eigen::Matrix<double, kMatrixSize, 1> x = matrix_NN.inverse() * v_Nd;
+    std::cout << "time of normal inverse is " <<
+    1000 * (clock() - time_stt) / static_cast<double>(CLOCKS_PER_SEC) << "ms" << std::endl;
```

Comment:
这里还有一个简单的写法是：
```suggestion
    1000.0 * (clock() - time_stt) / CLOCKS_PER_SEC << "ms" << std::endl;
```
抱歉之前看漏了。